packages/codebytes/src/editor.tsx

```diff
@@ -1,10 +1,164 @@
-import React from 'react';
+import {
+  FillButton,
+  FlexBox,
+  Spinner,
+  TextButton,
+  ToolTip,
+} from '@codecademy/gamut';
+import { CopyIcon } from '@codecademy/gamut-icons';
+import { theme } from '@codecademy/gamut-styles';
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+
+import type { languageOption } from './consts';
+import { Drawers } from './drawers';
+
+const Output = styled.pre<{ hasError: boolean }>`
+  width: 100%;
+  height: 20rem;
+  margin: 0;
+  padding: 0 1rem;
+  font-family: Monaco;
+  font-size: 0.875rem;
+  overflow: auto;
+  ${({ hasError }) => `
+  color: ${hasError ? theme.colors.orange : theme.colors.white};
+  background-color: ${theme.colors['gray-900']};
+`}
+`;
+
+const CopyIconStyled = styled(CopyIcon)`
+  margin-right: 0.5rem;
+`;
+
+const DOCKER_SIGTERM = 143;
 
 type EditorProps = {
+  hideCopyButton: boolean;
+  language: languageOption;
   text: string;
-  onChange: (text: string) => void;
+  // eslint-disable-next-line react/no-unused-prop-types
+  onChange: (
+    text: string
+  ) => void /* TODO: Add onChange behavior in DISC-353 */;
+  onCopy?: (text: string, language: string) => void;
 };
 
-export const Editor: React.FC<EditorProps> = ({ text, onChange }) => {
-  return <textarea value={text} onChange={(e) => onChange(e.target.value)} />;
+interface Response {
+  stderr: string;
+  stdout: string;
+  exit_code: number;
+}
+
+export const Editor: React.FC<EditorProps> = ({
+  language,
+  text,
+  hideCopyButton,
+  onCopy,
+}) => {
+  const [output, setOutput] = useState('');
+  const [status, setStatus] = useState<'ready' | 'waiting' | 'error'>('ready');
+  const [isCodeByteCopied, setIsCodeByteCopied] = useState(false);
+  const onCopyClick = () => {
+    if (!isCodeByteCopied) {
+      navigator.clipboard
+        .writeText(text)
+        // eslint-disable-next-line no-console
+        .catch(() => console.error('Failed to copy'));
+      onCopy?.(
+        text,
+        language
+      ); /* TODO: pass in onCopyBBCodeblock behavior from the future version we migrate to Next.js  */
```

Comment:
I learned what BBCode is here: https://en.wikipedia.org/wiki/BBCode

Reply:
oh cool!
can we link to the ticket that will handle this?

Reply:
yep! i think i'll ticket this out separately as i give proper consideration to the next.js implementation! can add the tickets first before i merge this pr

Reply:
sounds good!

Reply:
this ticket is here, though I removed the comment from this pr, since all the behavior should be captured in next.js. we're just consuming the onCopy behavior in this repo, whatever it is:

https://codecademy.atlassian.net/browse/DISC-402